packages/ember-data/lib/system/store.js

```diff
@@ -331,6 +331,10 @@ Store = Service.extend({
     // Set the properties specified on the record.
     record.setProperties(properties);
 
+    record.eachRelationship(function(key, descriptor) {
+      record._relationships[key].setHasData(true);
```

Comment:
@pangratz this is intentional, "future proofing"... we should probably add the parameter to `setHasData()` method though, even if we currently only support setting it to true.